code-challenges.test.js

```diff
@@ -10,51 +10,142 @@
 
-var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+  var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+  // Expected output: [-9, 7, 9, 199]
 
-var numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
+  var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+  // Expected output: [-823, 7, 23]
 
-var numbersToAdd3 = []
-// Expected output: []
+  // b) Create the function that makes the test pass.
+  // Creating a function that takes in an array as a parameter and returns a new array of only odd numbers sorted from least to greatest.
 
+  // Creating a function that takes in an array as a parameter
+  const oddNumbersLeastToGreatest = ([]) => {
 
+    // Variable to store the new array
+    let newArray = []
 
-// b) Create the function that makes the test pass.
+    // Using the higher-order function .map() to iterate through the array, and return a new array of only odd numbers
+    return array.map(value, index, array)
+  }
+
+
+  // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+
+  // a) Create a test with expect statements for each of the variables provided.
+  
+  // Adding a describe method that lists the name of the function
+  describe("accumulatingSum", () => {
+
+    // Adding a test method that describes what a function does
+    it("Takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.", () => {
+
+      // Adding an expect method which calls on the accumulatingSum() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+      expect(accumulatingSum([2, 4, 45, 9])).toEqual([2, 6, 51, 60])
+      expect(accumulatingSum([0, 7, -8, 12])).toEqual([0, 7, -1, 11])
+      expect(accumulatingSum([])).toEqual([])
+    })
+  })
+
+  // Test fails with ReferenceError: accumulatingSum is not defined
+
+  var numbersToAdd1 = [2, 4, 45, 9]
+  // Excpected output: [2, 6, 51, 60]
+
+  var numbersToAdd2 = [0, 7, -8, 12]
+  // Expected output: [0, 7, -1, 11]
+
+  var numbersToAdd3 = []
+  // Expected output: []
+
+  // b) Create the function that makes the test pass.
+
+  // Creating a function that takes in an array as a parameter
+  const accumulatingSum = (array) => {
+
+    // Using the .map() method to loop through the array and will return the empty array.
+    return array.map((value) => value + i++)
+  }
+}
+
+
+// I have to time-box this so I'm not sleep-deprived for the rest of the week, so this is my best shot at the moment. I worked on this for around 3 hours, I feel like I made a little progress. Some ideas are clicking a little better. I took a look at other classmates code to see if I understood how they did it. Some of it made sense, and some not. I think its just a matter of doing consistent practice every day for a few weeks.
```

Comment:
Yeah you have a lot of good stuff here but it seems like you have multiple tools flying around the shop here. Perhaps code snippets flashcards will help here a bit. 

Your tests look amazing!